src/dao/dependency/npm.js

```diff
@@ -0,0 +1,80 @@
+const BaseDao = require('~/src/dao').BaseDao
+const NpmDependencies = require('~/src/models/dependencies/Npm')
+const cleanModel = require('windbreaker-service-util/models/util/cleanModel')
+
+let dao
+
+/**
+ * Helper function for generating query strings
+ *
+ * @param { Object } knex - the knex instance
+ * @param { String } key - the key to look for
+ * @param { Field } field - the jsonb field to search
+ *
+ * NOTE: field needs to be "manually" injected into the template field because
+ * knex will wrap the value in quotes if handled injected normally (thus causing
+ * the query to be malformed
+ */
+function _jsonKeyExistsIn (knex, key, field) {
+  return knex.raw(`${field}::jsonb ->> ? is not null`, [ key ])
```

Comment:
Is this type cast really necessary? Would simply ```knex.raw(`${field} ->> ? is not null`, [ key ])``` work as well?

On a side note, it looks like knex's query building support around json is a bit lacking :(

Reply:
Nope not needed, sorry thought that I needed to be explicit